query_modules/schema.cpp

```diff
@@ -92,31 +92,65 @@ void Schema::ProcessPropertiesRel(mgp::Record &record, const std::string_view &t
   record.Insert(std::string(kReturnMandatory).c_str(), mandatory);
 }
 
+// Custom equality operator for vector of strings
+// We need this because Label1:Label2 and Label2:Label1 are the same
+struct VectorComparator {
+  bool operator()(const std::vector<std::string> &lhs, const std::vector<std::string> &rhs) const {
+    // Sort and compare the vectors of strings
+    std::vector<std::string> sortedLhs = lhs;
+    std::vector<std::string> sortedRhs = rhs;
+    std::sort(sortedLhs.begin(), sortedLhs.end());
+    std::sort(sortedRhs.begin(), sortedRhs.end());
+
+    return sortedLhs < sortedRhs;
+  }
+};
+
 void Schema::NodeTypeProperties(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
   mgp::MemoryDispatcherGuard guard{memory};
-  ;
   const auto record_factory = mgp::RecordFactory(result);
   try {
-    const mgp::Graph graph = mgp::Graph(memgraph_graph);
-    for (auto node : graph.Nodes()) {
+    std::map<std::vector<std::string>, std::tuple<std::set<std::string>, std::string, mgp::List>, VectorComparator>
+        node_types;  // map of node types, key is vector of labels, value is tuple of set of properties, type of node
+                     // and list of labels
+    std::unordered_map<std::string, mgp::Value> properties;
+    for (auto node : mgp::Graph(memgraph_graph).Nodes()) {
       std::string type = "";
       mgp::List labels = mgp::List();
+      std::vector<std::string> labels_vector = {};
       for (auto label : node.Labels()) {
         labels.AppendExtend(mgp::Value(label));
         type += ":`" + std::string(label) + "`";
+        labels_vector.emplace_back(label);
+      }
+
+      if (node_types.find(labels_vector) == node_types.end()) {
+        node_types[labels_vector] = std::make_tuple(std::set<std::string>({""}), type, labels);
       }
 
       if (node.Properties().size() == 0) {
-        auto record = record_factory.NewRecord();
-        ProcessPropertiesNode<std::string>(record, type, labels, "", "", false);
         continue;
       }
 
       for (auto &[key, prop] : node.Properties()) {
-        auto property_type = mgp::List();
+        auto tuple = node_types.at(labels_vector);
+        std::get<0>(tuple).insert(key);
+        std::get<0>(tuple).erase("");
+        node_types[labels_vector] = tuple;
+        properties[key] = prop;
+      }
+    }
+
+    for (auto &[labels, props] : node_types) {
+      for (auto const &prop : std::get<0>(props)) {
         auto record = record_factory.NewRecord();
-        property_type.AppendExtend(mgp::Value(TypeOf(prop.Type())));
-        ProcessPropertiesNode<mgp::List>(record, type, labels, key, property_type, true);
+        if (prop == "") {
+          ProcessPropertiesNode<std::string>(record, std::get<1>(props), std::get<2>(props), "", "", false);
```

Comment:
Can you make 1, and 2 some static constexpr constant which will explain what is exactly 1 and 2